### ISSUE TITLE
[Bug]: `jsx-curly-brace-presence` doesn't trigger on strings containing a quote character

### ISSUE THREAD

**Issue by tylerlaprade** [jsx-eslint/eslint-plugin-react#3795]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

The curly braces are unnecessary here and should be reported by the rule. If the single quote is replaced with any other character, the rule does trigger.
```jsx
<Foo
  bar={"'"}
/>
```

### Expected Behavior

Report an error with a suggested fix to
```jsx
<Foo
  bar="'"
/>
```

### eslint-plugin-react version

4.3.1

### eslint version

8.57.0

### node version

20.14.12
